adding the what-is-a-vector-database blog assets (#552)
qdrant/landing_page: qdrant-landing/content/blog/what-is-a-vector-database.md

Traditional databases, even with extensions that provide some vector handling capabilities, struggle with the complexities and demands of high-dimensional vector data. 

Handling of vector data is extremely resource-intensive. A traditional vector is around 6Kb. You can see how scaling to millions of vectors can demand substantial system memory and computational resources. Which is at least very challenging for traditional [OLTP](https://www.ibm.com/topics/oltp) and [OLAP](https://www.ibm.com/topics/olap) databases to manage.

![](/blog/what-is-a-vector-database/Why-Use-Vector-Database.jpg)

Vector databases allow you to understand the **context** or **conceptual similarity** of unstructured data by representing them as **vectors**, enabling advanced analysis and retrieval based on data similarity.

For example, in recommendation systems, vector databases can analyze user behavior and item characteristics to suggest products or content with a high degree of personal relevance. 

> In search engines and research databases, they enhance the user experience by providing results that are** semantically** similar to the query, rather than relying solely on the exact words typed into the search bar.

Reviewer: ```suggestion
In search engines and research databases, they enhance the user experience by providing results that are **semantically** similar to the query. They do not rely solely on the exact words typed into the search bar.
```

> If you're new to the vector search space, this article will get you started by explaining the key concepts and relations you need to get familiar with to fully understand how vector database systems work.

Reviewer: ```suggestion
If you're new to the vector search space, this article explains the key concepts and relationships that you need to know.
```

## Core Functionalities of Vector Databases

### What is Indexing?

Have you ever tried to find a specific face in a massive crowd photo? Well, vector databases face a similar challenge when dealing with tons of high-dimensional vectors. 

Now, imagine dividing the crowd into smaller groups based on hair color, then eye color, then clothing style. Each layer gets you closer to who you’re looking for. Vector databases use similar **multi-layered** structures called indexes to organize vectors based on their "likeness." 

This way, finding similar images becomes a quick hop across related groups, instead of scanning every picture one by one.

![](/blog/what-is-a-vector-database/Indexing.jpg)

Different indexing methods exist, each with its strengths. [HNSW](https://qdrant.tech/articles/filtrable-hnsw/) balances speed and accuracy like a well-connected network of shortcuts in the crowd. Others, like IVF or Product Quantization, focus on specific tasks or memory efficiency.

#### What is Binary Quantization?

Quantization is a technique used for reducing the total size of the database. It works by compressing vectors into a more compact representation at the cost of accuracy.

> [Binary Quantization](https://qdrant.tech/articles/binary-quantization/) is a fast indexing and data compression method available in Qdrant that allows for rapid vector comparison, which can speed up query processing times dramatically (up to 40x faster!).

Reviewer: ```suggestion
[Binary Quantization](https://qdrant.tech/articles/binary-quantization/) is a fast indexing and data compression method used by Qdrant. It supports vector comparisons, which can dramatically speed up query processing times (up to 40x faster!).
```

> Think of each data point as a ruler. Binary quantization splits this ruler in half at a certain point, marking everything above as "1" and everything below as "0". This binarization process results in a string of bits, representing the original vector.

Reviewer: As a reference for newer users, I wanted to find a link... this is the best I could do, you're welcome to substitute an alternative:

```suggestion
Think of each data point as a ruler. Binary quantization splits this ruler in half at a certain point, marking everything above as "1" and everything below as "0". This [binarization](https://deepai.org/machine-learning-glossary-and-terms/binarization) process results in a string of bits, representing the original vector.
```

The search then moves down progressively narrowing down to more closely related vectors. The goal is to narrow down the dataset to the most relevant items. The image below illustrates this.

![](/blog/what-is-a-vector-database/Similarity-Search-and-Retrieval)

Once the closest vectors are identified at the bottom layer, these points translate back to actual data, like images or music, representing your search results.

### Scalability

Vector databases often deal with datasets that comprise billions of high-dimensional vectors. This data isn't just large in volume but also complex in nature, requiring more computing power and memory to process. Scalable systems can handle this increased complexity without performance degradation. This is achieved through a combination of a **distributed architecture**,** dynamic resource allocation**, **data partitioning**,** load balancing**, and **optimization techniques**. 

> Systems like Qdrant, developed in Rust, exemplify scalability in vector databases by leveraging Rust's efficiency in **memory management** and **performance**, allowing handling of large-scale data with optimized resource usage.

Reviewer: ```suggestion
Systems like Qdrant exemplify scalability in vector databases. It leverages Rust's efficiency in **memory management** and **performance**, which allows handling of large-scale data with optimized resource usage.
```

### Efficient Query Processing 

The key to efficient query processing in these databases is linked to their **indexing methods**, which enable quick navigation through complex data structures. By mapping and accessing the high-dimensional vector space, HNSW and similar indexing techniques significantly reduce the time needed to locate and retrieve relevant data.

![](/blog/what-is-a-vector-database/search-query.jpg)

Other techniques like **handling computational load **and **parallel processing** are used for performance, especially when managing multiple simultaneous queries. Complementing them, **strategic caching** is also employed to store frequently accessed data, facilitating a quicker retrieval for subsequent queries.

### Using Metadata and Filters

Filters use metadata to refine search queries within the database. For example, in a database containing text documents, a user might want to search for documents not only based on textual similarity but also filter the results by publication date or author.

> When a query is made, the system can use **both** the vector data and the metadata to process the query. This means that the database doesn’t just look for the closest vectors but also considers the additional criteria set by the metadata filters, creating a more customizable search experience.

Reviewer: ```suggestion
When a query is made, the system can use **both** the vector data and the metadata to process the query. In other words, the database doesn’t just look for the closest vectors. It also considers the additional criteria set by the metadata filters, creating a more customizable search experience.
```

![](/blog/what-is-a-vector-database/metadata.jpg)

### Data Security and Access Control

Vector databases often store sensitive information. This could include personal data in customer databases, confidential images, or proprietary text documents. Ensuring data security means protecting this information from unauthorized access, breaches, and other forms of cyber threats.

> At Qdrant, this includes mechanisms like user authentication, role-based access control, attribute-based access control, encryption for data at rest and in transit, keeping audit trails, and advanced database monitoring and anomaly detection

Reviewer: ```suggestion
At Qdrant, this includes mechanisms such as:

 - User authentication
 - Role-based access control
 - Attribute-based access control
 - Encryption for data at rest and in transit
 - Keeping audit trails
 - Advanced database monitoring and anomaly detection
```
